lib/_io.scm

```diff
@@ -8343,6 +8343,167 @@
         (macro-dynamic-bind output-port port thunk)))))
 
 ;;;----------------------------------------------------------------------------
+(define-prim (##open-raw-device
+              direction
+              name
+              fd)
+  (define (fail)
+    (##fail-check-settings 1 ##open-raw-device direction name fd))
+
+  (##make-psettings
+   direction
+   '()
+   '()
+   fail
+   (lambda (psettings)
+     (let ((device
+            (##os-device-open-raw
```

Comment:
Actually, we can drop the function completely and only have `##make-raw-device-port`, where the caller does the open.